### PR TITLE
Fixed crash when trying to edit negative stock quantity

### LINKED ISSUE #2073: NumberFormatException: Invalid int: "-"
Sentry Issue: [WOOCOMMERCE-ANDROID-EA](https://sentry.io/organizations/a8c/issues/1567049669/?referrer=github_integration)

```
NumberFormatException: Invalid int: "-"
    at com.woocommerce.android.ui.products.ProductDetailViewModel.updateProductDraft(ProductDetailViewModel.kt:286)
    at com.woocommerce.android.ui.products.ProductDetailViewModel.updateProductDraft$default(ProductDetailViewModel.kt:274)
    at com.woocommerce.android.ui.products.ProductInventoryFragment$updateProductView$$inlined$with$lambda$3.invoke(ProductInventoryFragment.kt:125)
    at com.woocommerce.android.ui.products.ProductInventoryFragment$updateProductView$$inlined$with$lambda$3.invoke(ProductInventoryFragment.kt:24)
    at com.woocommerce.android.widgets.WCMaterialOutlinedEditTextView$setOnTextChangedListener$$inlined$doAfterTextChanged$1.afterTextChanged(TextView.kt:101)
...
(67 additional frame(s) were not displayed)

Invalid int: "-"
```

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #2073 by adding logic to support negative stock quantity value when editing products.

**This PR is in draft since it includes changes made in the `release/3.8` branch and should be merged into `develop` for the 3.9 beta release only when the 3.8 branch has been merged into `develop`.** cc @jkmassel 

#### To test
- Open `wp-admin` for a store and edit the stock quantity of a product to -5.
- Open the app and click on the `Products` tab.
- Click on the product you edited in step 1.
- Click on the `Inventory` section.
- Notice the stock quantity displayed is -5.
- Click on the stock quantity and remove the 5 from the text field.
- Notice the app no longer crashes. 
- Enter another amount such as 2 and click on `Done`.
- Verify that the stock quantity displayed is product detail is correct.
- It would also be good to test updating an empty stock quantity.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt [hunk 1]
```diff
@@ -2,6 +2,14 @@ package com.woocommerce.android.extensions
 
 import org.apache.commons.text.StringEscapeUtils
 
+/**
+ * Checks if a given string is a number (supports positive or negative numbers)
+ */
+fun String?.isNumeric(): Boolean {
+    return if (this.isNullOrBlank()) false
+    else Regex("-?\\d+(\\.\\d+)?").matches(this)
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt]: This works, but there already exists a function that we could use: `this?.toIntOrNull()?.let { true } ?: false`. Just an idea, though.

Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/StringExt.kt]: Thanks for the suggestion! Fixed in 9a9c1fe

**Comment by anitaa1990** [comment]
@0nko, thanks for the review! I fixed the suggestion. I have updated the base branch to `develop` and once the `release/3.8` has been merged into `develop`, we can merge this PR to `develop` so that this fix is available in the 3.9 beta release and not merged directly into the current 3.8 release.
